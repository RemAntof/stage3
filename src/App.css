.appContainer {
  display: flex;
  width: 100%;
  overflow: hidden;
  position: relative;
}

.mainPage {
  flex-shrink: 0;
  width: 100%;
  transition:
    transform 0.3s ease-in-out,
    width 0.3s ease-in-out;
}

#outlet {
  position: fixed;
  top: 0;
  right: 0;
  width: 0;
  height: 100%;
  transition: width 0.3s ease-in-out;
  background-color: #242424;
}

#outlet.visible {
  width: fit-content;
}

.mainPage.shifted {
  width: 60%;
}
