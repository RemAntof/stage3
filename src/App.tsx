import React, { useEffect, useState, useRef } from 'react';
import './App.css';
import MainPage from '@views/mainPage/MainPage';
import {
  Outlet,
  useLocation,
  useNavigate,
} from 'react-router-dom';

const App: React.FC = () => {
  const location = useLocation();
  const [outletVisible, setOutletVisible] = useState(false);
  const outletRef = useRef<HTMLDivElement>(null);
  const navigate = useNavigate();

  useEffect(() => {
    setOutletVisible(location.pathname !== '/');
  }, [location]);

  const onClick = (
    event: React.MouseEvent<HTMLDivElement, MouseEvent>
  ) => {
    if (
      outletVisible &&
      outletRef.current &&
      !outletRef.current.contains(event.target as Node)
    ) {
      setOutletVisible(false);
      navigate(-1);
    }
  };

  return (
    <div className="appContainer" onClick={onClick}>
      <div
        className={`mainPage ${outletVisible ? 'shifted' : ''}`}
      >
        <MainPage />
      </div>
      <div
        id="outlet"
        ref={outletRef}
        className={outletVisible ? 'visible' : 'hidden'}
      >
        {outletVisible ? <Outlet /> : null}
      </div>
    </div>
  );
};

export default App;
