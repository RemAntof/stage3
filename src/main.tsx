import ReactDOM from 'react-dom/client';
import App from './App.tsx';
import './index.css';
import {
  createBrowserRouter,
  RouterProvider,
} from 'react-router-dom';
import ErrorPage from '@views/errorView/errorView.tsx';
import CardDetail from '@components/cards/cardDetails/CardDetail.tsx';
import React from 'react';
import { Provider } from 'react-redux';
import store from '@services/store/store.ts';

const router = createBrowserRouter([
  {
    path: '/',
    element: <App />,
    errorElement: <ErrorPage />,
    children: [
      {
        path: ':animalName',
        element: <CardDetail />,
      },
    ],
  },
]);

ReactDOM.createRoot(
  document.getElementById('root')!
).render(
  <React.StrictMode>
    <Provider store={store}>
      <RouterProvider router={router} />
    </Provider>
  </React.StrictMode>
);
