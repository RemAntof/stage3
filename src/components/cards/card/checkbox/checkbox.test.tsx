import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { ThemeContext } from 'src/App';
import {
  useAppSelector,
  useAppDispatch,
} from '@hooks/useDispatchUseSelector';
import {
  addItem,
  removeItem,
} from '@components/flyout/cardsSlice';
import CheckBox from './checkbox';

jest.mock('@hooks/useDispatchUseSelector', () => ({
  useAppSelector: jest.fn(),
  useAppDispatch: jest.fn(),
}));

jest.mock('@components/flyout/cardsSlice', () => ({
  addItem: jest.fn(),
  removeItem: jest.fn(),
}));

describe('CheckBox Snapshot Tests', () => {
  const mockDispatch = jest.fn();

  beforeEach(() => {
    (useAppDispatch as jest.Mock).mockReturnValue(
      mockDispatch
    );
  });

  test('renders with light theme and checkbox unchecked', () => {
    (useAppSelector as jest.Mock).mockReturnValue({});

    const { asFragment } = render(
      <ThemeContext.Provider value="light">
        <CheckBox
          animal={{
            uid: 'ANMA0000005771',
            name: 'Andorian redbat',
            earthAnimal: false,
            earthInsect: false,
            avian: false,
            canine: false,
            feline: false,
          }}
        />
      </ThemeContext.Provider>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test('renders with dark theme and checkbox checked', () => {
    (useAppSelector as jest.Mock).mockReturnValue({
      ANMA0000005771: true,
    });

    const { asFragment } = render(
      <ThemeContext.Provider value="dark">
        <CheckBox
          animal={{
            uid: 'ANMA0000005771',
            name: 'Andorian redbat',
            earthAnimal: false,
            earthInsect: false,
            avian: false,
            canine: false,
            feline: false,
          }}
        />
      </ThemeContext.Provider>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test('toggles checkbox state and dispatches actions', () => {
    (useAppSelector as jest.Mock).mockReturnValue({});

    const { getByRole } = render(
      <ThemeContext.Provider value="light">
        <CheckBox
          animal={{
            uid: 'ANMA0000005771',
            name: 'Andorian redbat',
            earthAnimal: false,
            earthInsect: false,
            avian: false,
            canine: false,
            feline: false,
          }}
        />
      </ThemeContext.Provider>
    );

    const checkbox = getByRole('checkbox');

    fireEvent.click(checkbox);

    expect(mockDispatch).toHaveBeenCalledWith(
      addItem({
        uid: 'ANMA0000005771',
        name: 'Andorian redbat',
        earthAnimal: false,
        earthInsect: false,
        avian: false,
        canine: false,
        feline: false,
      })
    );

    fireEvent.click(checkbox);

    expect(mockDispatch).toHaveBeenCalledWith(
      removeItem({
        uid: 'ANMA0000005771',
        name: 'Andorian redbat',
        earthAnimal: false,
        earthInsect: false,
        avian: false,
        canine: false,
        feline: false,
      })
    );
  });
});
