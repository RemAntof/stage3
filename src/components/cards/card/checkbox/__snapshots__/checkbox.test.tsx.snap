// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBox Snapshot Tests renders with dark theme and checkbox checked 1`] = `
<DocumentFragment>
  <input
    checked=""
    class="checkBox checkBox dark"
    type="checkbox"
  />
</DocumentFragment>
`;

exports[`CheckBox Snapshot Tests renders with light theme and checkbox unchecked 1`] = `
<DocumentFragment>
  <input
    class="checkBox checkBox light"
    type="checkbox"
  />
</DocumentFragment>
`;
